From f81c6b5cded90323c3b756662d982597a2053693 Mon Sep 17 00:00:00 2001
From: Motohiro Kosaki <mkosaki@redhat.com>
Date: Wed, 10 Sep 2014 16:16:13 -0400
Subject: [x86] mm/numa: initialize numa_kernel_nodes in numa_clear_kernel_node_hotplug()

Message-id: <1410365775-5132-19-git-send-email-mkosaki@redhat.com>
Patchwork-id: 93494
O-Subject: [RHEL7 PATCH 18/20] arch/x86/mm/numa.c: initialize numa_kernel_nodes in numa_clear_kernel_node_hotplug()
Bugzilla: 810042
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

From: KOSAKI Motohiro <mkosaki@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=810042
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7914549
target RHEL version: 7.1
upstream status: merged
changes from upstream: none

commit 017c217a26e9bf6948482f751b30d0507e30a7d0
Author: Tang Chen <tangchen@cn.fujitsu.com>
Date:   Thu Feb 6 12:04:25 2014 -0800

    arch/x86/mm/numa.c: initialize numa_kernel_nodes in numa_clear_kernel_node_hotplug()

    On-stack variable numa_kernel_nodes in numa_clear_kernel_node_hotplug()
    was not initialized.  So we need to initialize it.

    [akpm@linux-foundation.org: use NODE_MASK_NONE, per David]
    Signed-off-by: Tang Chen <tangchen@cn.fujitsu.com>
    Tested-by: Gu Zheng <guz.fnst@cn.fujitsu.com>
    Reported-by: Dave Jones <davej@redhat.com>
    Reported-by: David Rientjes <rientjes@google.com>
    Tested-by: Dave Jones <davej@redhat.com>
    Cc: Ingo Molnar <mingo@elte.hu>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: "H. Peter Anvin" <hpa@zytor.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: KOSAKI Motohiro <mkosaki@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/mm/numa.c b/arch/x86/mm/numa.c
index 0e6efce..e34cb0a 100644
--- a/arch/x86/mm/numa.c
+++ b/arch/x86/mm/numa.c
@@ -560,7 +560,7 @@ static void __init numa_init_array(void)
 static void __init numa_clear_kernel_node_hotplug(void)
 {
  int i, nid;
- nodemask_t numa_kernel_nodes;
+ nodemask_t numa_kernel_nodes = NODE_MASK_NONE;
  unsigned long start, end;
  struct memblock_type *type = &memblock.reserved;
 
-- 
1.7.1